### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Progress2002**
### The following were implemented in this milestone:

- [x] Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Make sure that you used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
- [x] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [x] Think about how you can use two decorators in order to capitalize and trim people's names.
- [x] In the following steps you will get instructions for implementing the Decorator design pattern for this case.
- [x] Interface
- [x] Create a class Nameable.
- [x] Implement a method called correct_name that will raise a NotImplementedError.
- [x] Turn your Person class to Nameable
- [x] Make sure that your Person class inherits from Nameable
- [x] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- [x] Prepare base Decorator
- [x] Make sure that it inherits from Nameable.
- [x] In the constructor assign a nameable object from params to an instance variable.
- [x] Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- [x] Prepare CapitalizeDecorator and TrimmerDecorator
- [x] For the CapitalizeDecorator:
- [x] Create a class that inherits from the base Decorator class.
- [x] Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- [x] For the TrimmerDecorator:
- [x] Create a class that inherits from the base Decorator class.
- [x] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class BaseDecorator < Nameable
+  def initialize(nameable_obj)
+    super()
+    @nameable_obj = nameable_obj
+  end
+
+  def correct_name
+    @nameable_obj.correct_name
+  end
```
Comment (ichala) [inline base_decorator.rb]: - [ ]  your decorator class is missing` attr_accessor :nameable` kindly add it to match the requirements 🙏

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No Linter errors  ✔️ 


 ## Required Changes ♻️

- [ ] **_🛑 x_**
- [ ] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (Progress2002) [reply base_decorator.rb]: Thanks for your observation, the file has been updated.

**Review by binhussen — APPROVED** [review]
Hi @Progress2002 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Updated Request Changes ✔️
- Added Decorators ✔️
- Descriptive pull request ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**